src/Util/Security.php

```diff
@@ -0,0 +1,273 @@
+				return false;
+			} else {
+				$cid = 0;
+
+				if (!empty($_SESSION['remote'])) {
+					foreach ($_SESSION['remote'] as $visitor) {
+						if ($visitor['uid'] == $owner) {
+							$cid = $visitor['cid'];
+							break;
+						}
+					}
+				}
+
+				if (!$cid) {
+					return false;
+				}
+
+				$r = q("SELECT `contact`.*, `user`.`page-flags` FROM `contact` INNER JOIN `user` on `user`.`uid` = `contact`.`uid`
+					WHERE `contact`.`uid` = %d AND `contact`.`id` = %d AND `contact`.`blocked` = 0 AND `contact`.`pending` = 0
+					AND `user`.`blockwall` = 0 AND `readonly` = 0  AND ( `contact`.`rel` IN ( %d , %d ) OR `user`.`page-flags` = %d ) LIMIT 1",
+					intval($owner),
+					intval($cid),
+					intval(Contact::SHARING),
+					intval(Contact::FRIEND),
+					intval(Contact::PAGE_COMMUNITY)
+				);
+
+				if (DBA::isResult($r)) {
+					$verified = 2;
+					return true;
+				} else {
+					$verified = 1;
+				}
+			}
+		}
+
+		return false;
+	}
+
+	/// @TODO $groups should be array
+	public static function permissions_sql($owner_id, $remote_verified = false, $groups = null)
+	{
+		$local_user = local_user();
+		$remote_user = remote_user();
+
+		/**
+		 * Construct permissions
+		 *
+		 * default permissions - anonymous user
+		 */
+		$sql = " AND allow_cid = ''
+				 AND allow_gid = ''
+				 AND deny_cid  = ''
+				 AND deny_gid  = ''
+		";
+
+		/**
+		 * Profile owner - everything is visible
+		 */
+		if ($local_user && $local_user == $owner_id) {
+			$sql = '';
+		/**
+		 * Authenticated visitor. Unless pre-verified,
+		 * check that the contact belongs to this $owner_id
+		 * and load the groups the visitor belongs to.
+		 * If pre-verified, the caller is expected to have already
+		 * done this and passed the groups into this function.
+		 */
+		} elseif ($remote_user) {
+			/*
+			 * Authenticated visitor. Unless pre-verified,
+			 * check that the contact belongs to this $owner_id
+			 * and load the groups the visitor belongs to.
+			 * If pre-verified, the caller is expected to have already
+			 * done this and passed the groups into this function.
+			 */
+
+			if (!$remote_verified) {
+				if (DBA::exists('contact', ['id' => $remote_user, 'uid' => $owner_id, 'blocked' => false])) {
+					$remote_verified = true;
+					$groups = Group::getIdsByContactId($remote_user);
+				}
+			}
+
+			if ($remote_verified) {
+				$gs = '<<>>'; // should be impossible to match
+
+				if (is_array($groups)) {
+					foreach ($groups as $g) {
+						$gs .= '|<' . intval($g) . '>';
+					}
+				}
+
+				$sql = sprintf(
+					" AND ( NOT (deny_cid REGEXP '<%d>' OR deny_gid REGEXP '%s')
+					  AND ( allow_cid REGEXP '<%d>' OR allow_gid REGEXP '%s' OR ( allow_cid = '' AND allow_gid = '') )
+					  )
+					",
+					intval($remote_user),
+					DBA::escape($gs),
+					intval($remote_user),
+					DBA::escape($gs)
+				);
+			}
+		}
+		return $sql;
+	}
+
+	public static function item_permissions_sql($owner_id, $remote_verified = false, $groups = null)
```

Comment:
Since this method is directly related to the `item` table, please move it to `Model\Item::getPermissionsSQLByUserId`.